smellie/sepia/slm.py

```diff
@@ -3,138 +3,122 @@
 
 @raise_on_error_code
 def get_intensity_fine_step(dev_id, slot_id):
-    """get_intensity_fine_step(dev_if, slot_id)
-    Get the current intensity value of a fiven SLM driver module in 
-    per mille of the laser head controlling voltage
+    """
+    Get the current intensity value of a given SLM driver module, in 1/1000th increments of the laser head's control voltage
 
-    :param dev_id: the sepia device number (ordered from 0)
-    :type dev_id:  int
+    :param dev_id: the SEPIA device number, ordered from 0
+    :type dev_id: int
+
+    :param slot_id: the slot number, between 000 and 989
+    :type slot_id: int
 
-    :param slot_id: slot number, integer 000-989
-    :type slot_id:  int
-    
     :returns: intensity
-    :type intensity: int
+    :type intensity: double
     """
     intensity = ctypes.c_ushort()
     dll.SEPIA2_SLM_GetIntensityFineStep(dev_id, slot_id, intensity)
     return intensity
-    
-    
+
 @raise_on_error_code
 def set_intensity_fine_step(dev_id, slot_id, intensity):
-    """set_intensity_fine_step(dev_id, slot_id, intensity)
-    Set the intensity value of a given SLM driver module, in per mille
-    of laser head controlling voltage
+    """
+    Set the intensity value of a given SLM driver module, in 1/1000th increments of the laser head's control voltage
 
-    :param dev_id: the sepia device number (ordered from 0)
-    :type dev_id:  int
+    :param dev_id: the SEPIA device number, ordered from 0
+    :type dev_id: int
 
-    :param slot_id: slot number, integer 000-989
-    :type slot_id:  int
-    
-    :param intensity: 
-    :type intensity: int
+    :param slot_id: the slot number, between 000 and 989
+    :type slot_id: int
 
-    :returns: None
+    :param intensity: the requested laser head intensity
+    :type intensity: double
     """
     if not intensity in xrange(int(1e3)):
-        raise SepiaLogicError("set_intensity_fine_step: fine step is in per mille - must be 0 - 1000")
+        raise SepiaLogicError("Cannot set the intensity fine step - must be between 0 and 1000")
     dll.SEPIA2_SLM_SetIntensityFineStep(dev_id, slot_id, intensity)
 
-
 @raise_on_error_code
 def decode_head_type(head_type_code):
-    """decode_head_type(head_type)
-    Returns the head type string at list position head_type for any SLM 
-    module
-    
-    :param head_type_code: must be 0-3
+    """
+    Returns the head_type string at list position `head_type_code` for a given SLM module
+
+    :param head_type_code: must be between 0 and 3 inclusive
     :type head_type_code: int
 
     :returns: head_type
-    :type head_type: string    
+    :type head_type: string
+
+    :raises: :class:`.SepiaLogicError` if the head_type_code is invalid, i.e. not between 0 and 3	
     """
     if not head_type_code in xrange(4):
-        raise SepiaLogicError("decode_head_type:: head_type must be 0-3")
-
+        raise SepiaLogicError("Cannot decode head type - must be between 0 and 3")
     type = string_buffer()
```

Comment:
not a great idea to call this type - its a keyword in python. Looks like i did it though!